components/assistive-playwright-client/src/screenReaderClient.ts

```diff
@@ -0,0 +1,340 @@
+   * connected.
+   * @param url - Websocket URL such as `ws://127.0.0.1:7779/screen-reader`
+   */
+  static async create(url: string): Promise<ScreenReaderClient> {
+    const result = new ScreenReaderClient(url);
+    await new Promise((resolve, reject) => {
+      result._emitter.once("open", resolve);
+      result._emitter.once("error", reject);
+    });
+    return result;
+  }
+
+  private constructor(url: string) {
+    const socket = new Websocket(url);
+    this.socket = socket;
+    socket.onmessage = arg => {
+      const data = arg.data.toString("utf8");
+      this.messages.push(data);
+      this._emitter.emit("message", data);
+    };
+    socket.onopen = () => {
+      this.connected = true;
+      this._emitter.emit("open");
+    };
+    socket.onclose = () => {
+      this.connected = false;
+      this._emitter.emit("close");
+    };
+    socket.onerror = error => {
+      this._emitter.emit("error", error);
+    };
+  }
+
+  /**
+   * Disconnects the underlying websocket, thus preventing
+   * any further message from being received.
+   */
+  disconnect(): void {
+    if (this.connected) {
+      this.socket.close();
+    }
+  }
+
+  /**
+   * Execute the `condition` function on each received message until it returns a
+   * truthy value or the specified timeout occurs.
+   * If there is a timeout, the promise returned by this function is rejected.
+   * @param condition - function that is called for each message as long as it returns a falsy value. When it returns a
+   * truthy value, the process is stopped and that value is returned by
+   * waitForMessageCondition.
+   * @param options - `includeCurrent`: if true (the default), the `condition` function
+   * is first called for already received messages that are in {@link ScreenReaderClient.messages | messages}.
+   * Otherwise, it is only called on new messages, not yet received at the time `waitForMessageCondition` is called.
+   * `timeout`: the time in milliseconds to wait before the promise returned by
+   * `waitForMessageCondition` is rejected (if the condition does not return a
+   * truthy value during that time). Defaults to 10000ms (10s).
+   */
+  waitForMessageCondition<T>(
+    condition: (
+      value: string,
+      index: number,
+      messages: string[]
+    ) => T | undefined,
+    options: {
+      includeCurrent?: boolean;
+      timeout?: number;
+    } = {}
+  ): Promise<T> {
+    const { includeCurrent = true, timeout = 10000 } = options;
+    return new Promise<T>((resolve, reject) => {
+      if (includeCurrent) {
+        for (let msg = this.messages, i = 0, l = msg.length; i < l; i++) {
+          const result = condition(msg[i], i, msg);
+          if (result) {
+            resolve(result);
+            return;
+          }
+        }
+      }
+      const removeListeners = () => {
+        clearTimeout(plannedTimeout);
+        this.off("message", listener);
+        this.on("error", errorListener);
```

Comment:
Not sure if it's a typo or not here (there is a lot of code in this PR !), but is it not supposed to be : this.**off**("error", errorListener); ?

Reply:
Yes, thank you! I will fix it!